src/tasks/node/NodeTaskHelper.ts

```diff
@@ -109,7 +101,19 @@ export class NodeTaskHelper implements TaskHelper {
         const runOptions = runDefinition.dockerRun;
 
         const packagePath = NodeTaskHelper.inferPackagePath(helperOptions && helperOptions.package, context.folder);
-        const packageName = await NodeTaskHelper.inferPackageName(packagePath);
+
+        let nodePackage: NodePackage;
+
+        const getNodePackage: () => Promise<NodePackage> =

```

Comment:
Maybe we should move Lazy into utils, and create an additional AsyncLazy.

Reply:
Moved `Lazy<T>` to reuse here.  Didn't go so far as to create `AsyncLazy<T>`--a PR for another day.